core/common/src/main/java/alluxio/underfs/options/UnderFileSystemCreateOptions.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.options;
+
+import alluxio.Configuration;
+import alluxio.Constants;
+import alluxio.annotation.PublicApi;
+import alluxio.security.authorization.PermissionStatus;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * Method options for creating a file in UnderFileSystem.
+ */
+@PublicApi
+@NotThreadSafe
+public final class UnderFileSystemCreateOptions {
+  // Permission status to set for the file being created.
+  private PermissionStatus mPermissionStatus;
+  // Block size in bytes for HDFS files only, not applicable for other Under FileSystems.
+  private long mBlockSizeByte;
+
+  /**
+   * @return the default {@link UnderFileSystemCreateOptions}
+   */
+  public static UnderFileSystemCreateOptions defaults() {
```

Comment:
it seems odd to provide `defaults()` and a constructor ... it is mixing two idioms

we should either have two constructors or two defaults (`defaults()` and `defaults(conf)`, which is what I prefer)